### ISSUE TITLE
Request for SSH forwarding functionality for raisimUnity (remote simulator process & local raisimUnity)

### ISSUE THREAD

**Issue by MarkFzp** [raisimTech/raisimLib#40]
Thank you for maintaining this awesome library. Have used it for a while, I want to discuss a common use case that is not supported in current version. 

Usually, computation / training of a robot in raisim is done on a remote server which is connected by SSH from local machine. In order to use raisimUnity to visualize the robot's locomotion, I should either (1) first download the related files and data to the local machine, and then launch the local raisimUnity, which is a hassle; or (2) use X11 forwarding of raisimUnity that is launched on the remote server, which results in sluggish visualization due to high bandwidth requirement. 

Instead, SSH port forwarding seems like a better option, since raisimUnity and the simulator process is connected by using TCP/IP. I tried this setting with a remote simulator process and a local raisimUnity, but I got the following error message in raisimUnity regarding failed resources loading: 

![image](https://user-images.githubusercontent.com/23516384/105253549-b576c000-5b4d-11eb-9ed9-37d4bced3daf.png)

**Comment by jhwangbo** [comment]
Can you elaborate on your setup? Do you have a remote Linux machine and a local Mac?

I haven't tried it between two different OS's. My students use a remote Linux machine and a local Linux machine and it works well. I'll investigate it on weekend.

**Comment by jhwangbo** [comment]
and does the same simulation/visualization work well on a local machine?

**Comment by MarkFzp** [comment]
The above screenshot which shows the error msg is taken on the raisimUnity on my local Mac (git commit 091b625f17f9dccafda17a5f1e4c22e06dfa71d5) when it's trying to try to connect to a remote Linux (git commit 5a6592d5937639d0084697a3bc88f353ed5ca14a) where the simulator runs. 

local Mac system info:
- macOS 10.15.7
- 8-Core Intel Core i9

remote Linux system info:
- Ubuntu 20.04.1 LTS (Focal Fossa)
- Intel(R) Core(TM) i9-10920X CPU @ 3.50GHz
 
--------

I also tested a different setting: raisimUnityOpenGL on a local Linux machine (git commit 33572211d00c83de09d914bb0ff8703fcbb5112e Nov 5 2020) and the simulator on a remote Linux machine (same one in the previous setting and git commit 5a6592d5937639d0084697a3bc88f353ed5ca14a). It failed with a resource loading error. Should local resources have the exact same absolute path as the ones on the remote machine?

![image](https://user-images.githubusercontent.com/23516384/105472120-88b7cb00-5c69-11eb-88f4-67f288d5641b.png)

local Linux system info:
- Ubuntu 18.04.5 LTS (Bionic Beaver)
- Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz

--------

In addition, I tried to update the library on the local linux machine to commit 1d833888c0679fc749dce8e7efdaf60c08a47062 (Jan 15 2021). Same remote linux machine and setting. The error msg of raisimUnityOpenGL changed to: 

![image](https://user-images.githubusercontent.com/23516384/105473018-9ae63900-5c6a-11eb-99df-33bb2a2e796e.png)

**Comment by jhwangbo** [comment]
Can you just update both systems to the latest one? RaiSimUnity is constantly improving and there is no guarantee that two different versions use the same communication protocol.

 "Cannot find the meshes.." error is a good sign. This means that you are connected and the two protocols (most likely) match. But you have to have meshes locally to visualize the simulation. On the very bottom, you can add resource directories that contain the mesh files (or its upper directory).

This is how we find the mesh file. https://github.com/raisimTech/raisimUnity/blob/ddc42b8de840c18cd994a7dc64034b161b33e1b9/Assets/Scripts/ResourceLoader.cs#L75
 
I'll close this issue. I guess I have to keep a better versioning system now in order to avoid such confusion.

**Comment by MarkFzp** [comment]
![image](https://user-images.githubusercontent.com/23516384/105646289-16173d00-5e6d-11eb-83c0-c5a94cc7285d.png)

I updated raisimLib to the latest version on both my local mac and remote Ubuntu server. RaisimUnity in both machines work perfectly when it pairs with a raisim simulator on the same machine. However, when I do ssh port forwarding of raisim simulator's visualization from remote ubuntu server to my local mac machine, I got this error message. The specs of both machines are listed in the previous posts.
